pkg/reportingplugins/mercury/mercury.go

```diff
@@ -181,7 +181,7 @@ type ReportCodec interface {
 type Fetcher interface {
 	// FetchInitialMaxFinalizedBlockNumber should fetch the initial max
 	// finalized block number from the mercury server.
-	FetchInitialMaxFinalizedBlockNumber(context.Context) (int64, error)
+	FetchInitialMaxFinalizedBlockNumber(context.Context) (*int64, error)
```

Comment:
nit/ did you consider using `-1` instead of `nil` to avoid the signature change?

Reply:
Yes but it's even more confusing because -1 is actually a valid value that is used already